### ISSUE TITLE
french dictionnary (apostrophe case)

### ISSUE THREAD

**Issue by menny** [AnySoftKeyboard/AnySoftKeyboard#109]
https://code.google.com/p/softkeyboard/issues/detail?id=573

The comment #7 explain everything : the apostrophe and hyphen should be concider as a normal word separator.

**Comment by Evpok** [comment]
Any news on this? I didn't find the French language pack sources, but in the store version the issue is still there.

**Comment by breversa** [comment]
On a related note, I'd love to build a newer/better version of the french dictionnary. I find the current one lacking in many ways ; for instance, several common conjugations are missing.

Any howto would be helpful ! :-)

**Comment by Evpok** [comment]
@brevera If you do, count me in.

**Comment by breversa** [comment]
I did a bit of research and here are my foundings so far :

- From https://f-droid.org/repository/browse/?fdfilter=french&fdid=com.anysoftkeyboard.languagepack.french_xlarge I understand that the French XL dictionnary is apparently based on the dicollecte.org dictionnary (http://dicollecte.org/home.php)

- However, this dictionnary only contains "root" words, e.g. infinitive verbs, singular and masculine forms of nouns, etc. It apparently does not include conjugated verbs or "accorded" nouns or adjectives.

- The dicollecte.org *lexique* (http://dicollecte.org/download.php?prj=fr), on the other hand, not only contains pretty much all the forms of verbs and nouns, but also elements of frequency.

- dicollecte lexique is published under the Mozilla Public Licence.

So I guess all that's left to do is build a french language pack based on dicollecte's lexique.

If only I could get my hands on that howto I found some day…

**Comment by menny** [comment]
That's very interesting!

What is your programming and Android Development experience?

**Comment by breversa** [comment]
Though I have a developer's degree (mainly Java, C and PHP), it dates
back to the early 2000s and I've never thought myself of being a
developper. So no real experience, and definitively no Android one,
but I guess I have a semi-functionnal brain and I like to find or even
sometimes provide solutions. :-)

What did you have in mind ?

2015-03-15 18:29 GMT+01:00 Menny Even Danan <notifications@github.com>:

**Comment by breversa** [comment]
… and here's the building guide I was thinking of : https://code.google.com/p/softkeyboard/wiki/BinaryDictionaries :-)

**Comment by menny** [comment]
This SO outdated!

Basically, https://github.com/AnySoftKeyboard/LanguagePack is the base of language packs (see branches for concrete implementations)  
But, I doubt it works right now, because all of the dependencies have changed.

You can have a go at that 😃

**Comment by breversa** [comment]
Yeah, that's what I feared and was about to ask.

I guess the least I can do is provide a french words.xml file with dicollecte's data. Would that be enough for you (or anyone else) to update the French language pack ?

And BTW, do the <w> elements in words.xml need to be ordered according to their frequency (= "f" value) or not ?

**Comment by menny** [comment]
Yes, words XML is good enough. Attach it here.
And, yes, `f` is the frequency

**Comment by breversa** [comment]
I'm still a beginner regarding GitHub, so… how do you attach files to an issue comment ?

**Comment by Evpok** [comment]
If you are looking for an extensive dictionary, the [Lefff](http://alpage.inria.fr/~sagot/lefff.html) is the most complete there is. It doesn't have frequency data, though.

**Comment by breversa** [comment]
Nevermind, I think I found a solution : https://my.owndrive.com/public.php?service=files&t=8e5c75acbf7e8766b2eb6efb09d24fa7

And here is the small awk script I wrote to generate the above file : https://my.owndrive.com/public.php?service=files&t=3486b43d01e665b80f25c9d62f7f1007

**Comment by breversa** [comment]
Thanks Evpok. I've had a (lightning-)quick look at the LEFFF.

However, I don't think an extensive dictionnary is the best thing for a phone dictionnary : it's the most common words that are needed, not necessarily the most exotic ones. YMMV, though, if you have specific needs.

**Comment by Evpok** [comment]
Just saying :) However it still doesn't help with our tokenisation problem.

**Comment by breversa** [comment]
Oops, yeah, sorry about hijacking the initial issue… :-/

**Comment by breversa** [comment]
Hi Menny, I just wanted to know if the fr.xml file I provided 3 posts above was enough for you to generate a new french dictionnary ?

**Comment by xavihernandez** [comment]
#540

**Comment by djibux** [comment]
Hello.  Is someone currently looking into that issue?

**Comment by breversa** [comment]
I'm afraid no one is... 😞
Le 15 déc. 2015 11:04, djib <notifications@github.com> a écrit :Hello.  Is someone currently looking into that issue?

**Comment by xavihernandez** [comment]
Yep strange.
By the way, lots of keyboard properly support French, especially AOSP one which is open source.

**Comment by breversa** [comment]
Yeah, and more and more even support the bépo layout now (and with
better/correct dictionnary) ! I henceforth have less and less use for
AnySoft Keyboard. :-(

2015-12-15 12:36 GMT+01:00 Eric Mourougandy <notifications@github.com>:

**Comment by djibux** [comment]
AnySoft has support for bépo (event though I tend to prefer azerty on touch devices).

I can't seem to find the AOSP keyboard on f-droid :(

**Comment by breversa** [comment]
Yes, that's why I've been using ASK so far. But now that other keyboards
(AOSP in CM12+, Swiftkey, probably more) do, I don't have any real reason…

2015-12-15 14:41 GMT+01:00 djib <notifications@github.com>:

**Comment by xavihernandez** [comment]
@djibux I dont think it is built by F-Droid. 
When using CyanogenMod ROM you have AOSP keyboard by default. I used it before, but it is too tall, and there is no way t disabled numbers keys row (which was not here before).

**Comment by djibux** [comment]
Thanks @xavihernandez. Very unfortunately my device isn't supported by CyanogenMod.  I use a stock version of android that I have degooglized myself.  Google Keyboard was the available keyboard.

**Comment by xavihernandez** [comment]
@djibux  Oh yeah. Well, if you use Google Keyboard you will have an idea about AOSP keyboard because they have the same base. I don't really know what is closed-source in Google Keyboard tho...

**Comment by xavihernandez** [comment]
@menny  thanks, hope it get integrated soon =)

**Comment by breversa** [comment]
Hi @menny, thanks for looking into this. :-)

Shall I provided an updated version of the fr.xml file ?

**Comment by parawizard** [comment]
Been awhile since I checked back here. Cool stuff! I been too busy working for a long time now (: 

Good luck with the enhancement!

**Comment by homlett** [comment]
That would be amazing that this issue would be solved!

Idk if it's the right place, but about the French keyboard lacks, guillemets (the French quotes style: « and ») are missing also.
https://en.m.wikipedia.org/wiki/Guillemet

**Comment by stragu** [comment]
@homlett The azerty keyboard has them with a long press on the "/' key, but having the French guillemets by default would probably be best (i.e. a «/'/» key). And maybe I'm a dreamer, but ideally it would automatically insert an unbreakable space after "«" and before "»" 😉

**Comment by homlett** [comment]
I don't find them... I'm using AnySoftkeyBoard 1.8.195 and the French pack 20111029 (both from F-Druid).

![](https://framapic.org/oERT7raxM83x/3j7XF91SdgjO)


Otherwise, you're right about the french style quotes key and non-breaking spaces. That would be very nice (like swipes, but it's another issue!).

**Comment by stragu** [comment]
Ok I figured out what the difference is: we have the exact same software versions, but I think that you are using the regular "common bottom generic row" whereas I am using the "new generation - testing" (go to User interface > Even more... > Common bottom generic row).

And to clarify my previous comment: I was not talking about the **/** key, but rather using the character as a separator :)

**Comment by homlett** [comment]
You're right! Thanks a lot, it's much better now! Btw, the guillemets are also available with the qwerty keyboard.
Have a good day!

**Comment by papjul** [comment]
I’m working on an updated French keyboards + dictionary and I’m running into this issue.
Is ASK based on AOSP keyboard? Because if so, this is how Google is handling different typography in different languages:
- https://android.googlesource.com/platform/packages/inputmethods/LatinIME/+/master/java/res/values-fr/donottranslate-config-spacing-and-punctuations.xml
- https://android.googlesource.com/platform/packages/inputmethods/LatinIME/+/master/java/src/com/android/inputmethod/latin/settings/SpacingAndPunctuations.java
Though I have not enough Android knowledge to implement it in ASK…

**Comment by friesenkiwi** [comment]
See https://github.com/AnySoftKeyboard/LanguagePack/issues/12
Maybe the original issue (and some of the follow-ups here ;-)) is already solved with the new dictionary?
@papjul how far did you progress with your pack? Do you think, you can put up a PR as soon as the resurrected sources have been merged?

**Comment by papjul** [comment]
I have designed French/Belgian and Canadian keyboards (with accents versions), though I'm still waiting for a fix in another issue for long pressed characters.
I have a working updated Dicollecte dictionary. However, I tried with the AOSP dictionary and it is buggy, words with accentuated letters are not shown, we should not use it, but Dicollecte dictionary is better anyway so I don't really care for the moment. I also tried to merge AOSP english+AOSP french dictionaries for people using both languages like me, but since AOSP french doesn't work well, we can't use it. I can't merge AOSP english with Dicollecte french because frequency is not on the same scale.
I will release sources when I'm home but it is still very experimental.

**Comment by breversa** [comment]
Sorry, I wanted to update the fr.xml file yesterday night but got busy.

I guess the frequency can be converted from Dicollecte's 9-1 range, to ASK
255-1 using a scaling formula (not a math guy myself, but I'll figure
something out). Like multilplying by 28.333 (= 255/9) for frequencies > 1,
or something. ^^

2017-02-17 11:37 GMT+01:00 Julien Papasian <notifications@github.com>:

**Comment by papjul** [comment]
That's not how the frequency works in Dicollecte.
If you make a scale from 255 to 1, you will have the most popular in French at 255 and the second most popular word at 128, and so on until the 10th most popular at about 2 or 3. Because the frequency is calculated to the number of times it is used in French sentences. However, scale in AOSP keyboards don't work like this, this is far more linear.

**Comment by friesenkiwi** [comment]
Let's concentrate the discussion about a new French Dictionary at https://github.com/AnySoftKeyboard/LanguagePack/issues/12

**Comment by Silmathoron** [comment]
Yep, also having issue with this... could somebody tell us how we could contribute to improving this? Would sharing user dict be helpful or is this a matter of adding some kind of grammar rule in a parser?

**Comment by jlemonde** [comment]
I experience the same kind of issues, it is very annoying that the apostrophes are not correctly predicted. I would like that when I type a word by omitting the apostrophe, it recognises that there should be one, for instance *jai → j'ai*, *aujourdhui → aujourd'hui*, and so on. For now it does not. Is a fix possible ?

**Comment by PrSunflower** [comment]
Hello! Has the issue been solved?

I typed `m'avertir` correctly but the AZERTY keyboard autocorrected into `l'avertir`.

![Screenshot_20200814-162440_1](https://user-images.githubusercontent.com/59576952/90261334-18508680-de4d-11ea-9b6a-812bd3aec3a6.png)

**Comment by MayeulC** [comment]
Hello @PrSunflower =. No, it hasn't, unfortunately. You have probably typed "l'avertir" often enough that it is in your user dictionary, thus ASK suggests it.

**Comment by PrSunflower** [comment]
Hi @MayeulC ,

Oh too bad. Thanks for the suggestion, but I have checked my AZERTY dictionary and `l'avertir` is not in the list.

![Screenshot_20200815-111440_1](https://user-images.githubusercontent.com/59576952/90309442-fd335480-dee8-11ea-822a-c6e57f5eb3cd.png)
